transport/transport.go

```diff
@@ -75,10 +75,10 @@ type Transport interface {
 	// Listen listens on the passed multiaddr.
 	Listen(laddr ma.Multiaddr) (Listener, error)
 
-	// Protocol returns the set of protocols handled by this transport.
-	//
+	// Protocols returns the set of protocols handled by this transport.
+	// If protocols A and B are returned, this means that this transport supports running B on top of A.
```

Comment:
Is this a semantic change, or was the previous interpretation supposed to be stacked too?

Reply:
Oh wait, I just read the PR description. I think we will find this interface limiting, and suggest something more sophisticated. [][]protocol.ID could be an option, but we might want to go with a struct for more developer friendliness.

Reply:
That’s correct, it is a semantic change (as I’ve noted in the PR description). None of our transports ever returned more than one value here though.

Reply:
Just read your 2nd comment after posting my reply. We could define a `ProtocolStack` as `[]Protocol`, and then return a `[]ProtocolStack` here.

One example where returning multiple protocols might be useful is when we mint a new code point for a new QUIC version. On the other hand, one might argue that if two code points are warranted, the differences between the two transports are probably large enough to just initializing two objects.

Reply:
That would be more self-descriptive, yes. Not sure if ProtocolStack is the right term --  since in the past we've talked about libp2p being a factory of _stacks_ with different characteristics and API shapes/semantics (e.g. message orientation). So that might get overloaded in the future. Perhaps ProtocolSeq?

Reply:
Also, the term "stack" to me denotes some kind of behaviour associated with that type, versus that term just being a keyword to enable AND and OR combinations of protocols.